contrib/pkg/createcluster/create.go

```diff
@@ -210,6 +216,14 @@ create-cluster CLUSTER_DEPLOYMENT_NAME --cloud=gcp --gcp-project-id=PROJECT_ID`,
 	flags.Int64Var(&opt.WorkerNodes, "workers", 3, "Number of worker nodes to create.")
 	flags.StringVar(&opt.ManifestsDir, "manifests", "", "Directory containing manifests to add during installation")
 
+	// Flags related to adoption.
+	flags.BoolVar(&opt.Adopt, "adopt", false, "Enable adoption mode for importing a pre-existing cluster into Hive. Will require additional flags for adoption info.")
+	flags.StringVar(&opt.AdoptAdminKubeConfig, "adopt-admin-kubeconfig", "", "Path to a cluster admin kubeconfig file for a cluster being adopted. (required if using --adopt)")
+	flags.StringVar(&opt.AdoptInfraID, "adopt-infra-id", "", "Infrastructure ID for this cluster's cloud provider. (required if using --adopt)")
+	flags.StringVar(&opt.AdoptClusterID, "adopt-cluster-id", "", "Cluster UUID used for telemetry. (required if using --adopt)")
```

Comment:
Is ClusterID required? We use it for deprovisioning by tag with the following comment. Is that comment still applicable?
```
// Also cleanup anything with the tag for the legacy cluster ID (credentials still using this for example)
```

Reply:
If you go back to 4.1 I *think* it's still applicable. The globally unique identifier for a cluster felt important enough to keep as required even if we could dance away from it, may prove useful in relating with other systems.

Reply:
I guess my question is more about whether it is important enough to **require** it for a dev-centric adopt feature. Or can we say a little extra work for a dev when creating an adopted cluster by allowing it to be omitted.

Reply:
My thought was yes, by requiring it we're making whoever is adopting think about it, if it's a dev we know we can fake it, anyone else using this would need to make that determination themselves, but I kinda like that this forces them to acknowledge it.

Reply:
Makes sense.